pcdsdevices/utils.py

```diff
@@ -198,13 +198,12 @@ def schedule_task(func, args=None, kwargs=None, delay=None):
             break
 
     def schedule():
-        if matched_thread is None:
+        if matched_thread is not None and context.event_thread is not None:
```

Comment:
Pesky `if` statements - this seems an important fix! Main source of the race condition you ran into, perhaps?

Reply:
Yes! The critical bug here was that it was possible for `schedule` to exit without running any code at all!